spec/views/user_show_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe 'users/index', type: :feature do
+  before(:each) do
+    @user = User.create(name: 'John Doe', photo: 'https://i.pravatar.cc/300',
+                        bio: 'Alien biologist', posts_counter: 1)
+    5.times do |i|
+      Post.create(title: "Post - #{i}", text: 'This is the first post',
+                  comments_counter: 5, likes_counter: 0, author_id: @user.id)
+    end
+    visit user_posts_path(id: @user.id)
+  end
+
+  it 'displays user profile photo' do
+    expect(page).to have_css("img[src*='https://i.pravatar.cc/300']")
+  end
+
+  it 'displays username' do
+    expect(page).to have_content(@user.name)
+  end
+
+  it 'displays number of posts' do
+    expect(page).to have_content('Number of Posts: 5')
+  end
+
+  it "displays the user's first 3 posts" do
+    expect(page).to have_content('Post - 1')
+    expect(page).to have_content('Post - 2')
+    expect(page).to have_content('Post - 3')
+  end
```

Comment:
Dear @jaimevillegas and @SFarida,
Good job for writing test cases However there're a few test cases that are missing: 
1. Tests that assert you can see the user's bio.
2. Tests that asserts you can see a button that lets me view all of a user's posts.
3. Tests that redirect to the post's show page when you click on a user's post.
4. Tests that redirect to the user's post's index page when you click on a **see all posts**.

You can get help from this cheatsheet as well: https://gist.github.com/zhengjia/428105

Reply:
Hello @zainsadaqat 

Thank you for your review! We have worked on solving that!